Sources/PackageCollections/PackageCollections.swift

```diff
@@ -81,34 +81,40 @@ public struct PackageCollections: PackageCollectionsProtocol {
             case .failure(let error):
                 callback(.failure(error))
             case .success(let sources):
-                let identifiers = sources.map { .init(from: $0) }.filter { identifiers?.contains($0) ?? true }
-                if identifiers.isEmpty {
-                    return callback(.success([]))
+                let identiferSource = sources.reduce(into: [PackageCollectionsModel.CollectionIdentifier: PackageCollectionsModel.CollectionSource]()) { result, source in
+                    result[.init(from: source)] = source
                 }
-                let collectionOrder = identifiers.enumerated().reduce([Model.CollectionIdentifier: Int]()) { partial, element in
-                    var dictionary = partial
-                    dictionary[element.element] = element.offset
-                    return dictionary
+                let identifiersToFetch = identiferSource.keys.filter { identifiers?.contains($0) ?? true }
+
+                if identifiersToFetch.isEmpty {
+                    return callback(.success([]))
                 }
-                self.storage.collections.list(identifiers: identifiers) { result in
+
+                self.storage.collections.list(identifiers: identifiersToFetch) { result in
                     switch result {
                     case .failure(let error):
                         callback(.failure(error))
                     case .success(var collections):
+                        let sourceOrder = sources.enumerated().reduce([Model.CollectionIdentifier: Int]()) { result, item in
```

Comment:
`reduce(into:` not applicable here?

Reply:
good catch. that was an oversight on my part.